src/AutoRest.CSharp/Mgmt/Decorator/InheritanceChooser.cs

```diff
@@ -146,28 +144,37 @@ private static bool IsEqual(System.Type parentType, ObjectTypeProperty[] propert
 
             foreach (var childProperty in childProperties)
             {
-                PropertyInfo? parentProperty;
-                CSharpType childPropertyType = childProperty.Declaration.Type;
-                if (parentDict.TryGetValue(childProperty.Declaration.Name, out parentProperty))
-                {
-                    if (parentProperty.PropertyType.IsGenericType)
-                    {
-                        if (!childPropertyType.Equals(new CSharpType(parentProperty.PropertyType)))
-                            return false;
-                    }
-                    else if (parentProperty.PropertyType.FullName != $"{childPropertyType.Namespace}.{childPropertyType.Name}" &&
-                        !IsAssignable(parentProperty.PropertyType, childPropertyType) &&
-                        !(parentProperty.PropertyType.IsGenericParameter && IsAssignable(parentProperty.PropertyType.BaseType!, childPropertyType)))
-                    {
-                        //TODO(ADO item 5712): deal with protected setter
-                        return false;
-                    }
-                }
-                else
-                {
+                if (!DoesPropertyExistInParent(childProperty, parentDict))
                     return false;
-                }
             }
+
+            return true;
+        }
+
+        private static bool DoesPropertyExistInParent(ObjectTypeProperty childProperty, Dictionary<string, PropertyInfo> parentDict)
+        {
+            PropertyInfo? parentProperty;
+            CSharpType childPropertyType = childProperty.Declaration.Type;
+
+            if (!parentDict.TryGetValue(childProperty.Declaration.Name, out parentProperty))
+                return false;
+
+            if (parentProperty.PropertyType.IsGenericType)
+            {
+                if (!childPropertyType.Equals(new CSharpType(parentProperty.PropertyType)))
+                    return false;
+            }
+            else if (parentProperty.PropertyType.FullName == $"{childPropertyType.Namespace}.{childPropertyType.Name}" ||
+                IsAssignable(parentProperty.PropertyType, childPropertyType))
+            {
+                if (childProperty.IsReadOnly != (parentProperty.GetSetMethod() == null))
```

Comment:
question, GetSetMethods will return not null as long as setter is defined right? Should we check != null but have private modifier?

Reply:
This will ignore private setters unless we use the overload and pass in true for nonPublic.